src/main/java/duke/io/Storage.java

```diff
@@ -0,0 +1,108 @@
+     * Tasks fields are delimited by ',', in the following format: <br>
+     * <pre> TodoTask Task: T ,Task Done Status ,Task Name. </pre>
+     * <pre> DeadlineTask Task: D, Task Done Status ,Task Name, Task Deadline Date. </pre>
+     * <pre> EventTask: E ,Task Done Status ,Task Name, Task Event Date. <br> </pre>
+     *
+     * @param folderName folder name to create the save file.
+     * @param fileName name of save file.
+     * @param arr list of tasks to be saved.
+     * @return 0: save success, -1: error encountered.
+     */
+    public static int saveFile(String folderName, String fileName, ArrayList<Task> arr) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/M/yyyy");
+        try {
+            Files.createDirectories(Paths.get(folderName));
+            String filePath = folderName + "/" + fileName;
+            File myObj = new File(filePath);
+            myObj.createNewFile();
+            FileWriter writer = new FileWriter(filePath);
+            StringBuilder sb = new StringBuilder();
+            for (Task t : arr) {
+                if (t.toString().charAt(1) == 'T') {
+                    sb.append(String.format("%s,%s,%s,",
+                            t.toString().charAt(1), t.isDone() ? "T" : "F",
+                            t.getTaskName()))
+                        .append(";");
+                } else if (t.toString().charAt(1) == 'D') {
+                    sb.append(String.format("%s,%s,%s,%s",
+                            t.toString().charAt(1),
+                            t.isDone() ? "T" : "F",
+                            t.getTaskName(), ((DeadlineTask) t).getDueDate().format(formatter)))
+                        .append(";");
+                } else if (t.toString().charAt(1) == 'E') {
+                    sb.append(String.format("%s,%s,%s,%s",
+                                    t.toString().charAt(1),
+                                    t.isDone() ? "T" : "F", t.getTaskName(), (
+                                            (EventTask) t).getDate().format(formatter)))
+                            .append(";");
+                }
+            }
+            sb.setLength((sb.length() - 1));
+            writer.write(sb.toString());
+            writer.close();
+            return 0;
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+            return -1;
+        }
+    }
+
+    /**
+     * Loads the content from the file specified into the specified TaskList.
+     *
+     * @param filePath path of the file to be read.
+     * @param tl TaskList to be updated.
+     * @return 0: load success, -1: error encountered.
+     */
+    public static int loadFile(String filePath, TaskList tl) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/M/yyyy");
+        try {
+            FileInputStream fs = new FileInputStream(filePath);
+            Scanner sc = new Scanner(fs);
+            StringBuilder ss = new StringBuilder();
```

Comment:
Consider using longer names because the 2 alphabet variables make it difficult to remember which is which when you have so many of them, like differentiating `ss` from `sc`.
```suggestion
            FileInputStream fileInput = new FileInputStream(filePath);
            Scanner scanner = new Scanner(fs);
            StringBuilder stringBuilder = new StringBuilder();
```